types/eslint/index.d.ts

```diff
@@ -725,7 +725,7 @@ export namespace Linter {
         extends?: string | string[] | undefined;
         globals?: { [name: string]: boolean | "readonly" | "readable" | "writable" | "writeable" } | undefined;
         noInlineConfig?: boolean | undefined;
-        overrides?: ConfigOverride[] | undefined;
+        overrides?: Array<ConfigOverride<Rules>> | undefined;
```

Comment:
I still think something like the following would make more sense?

```ts
interface BaseConfig<Rules extends RulesRecord = RulesRecord, OverrideRules extends RulesRecord = Rules> extends HasRules<Rules> {
  overrides?: Array<ConfigOverride<OverrideRules>> | undefined
}
```

Reply:
Ok done 614bb973554875f10ddf1cbcaadae2d283f54c37